pkg/controller/operators/catalog/operator.go

```diff
@@ -290,21 +297,60 @@ func (o *Operator) handleDeletion(obj interface{}) {
 	if !ok {
 		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
 		if !ok {
-			utilruntime.HandleError(fmt.Errorf("Couldn't get object from tombstone %#v", obj))
+			utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %#v", obj))
 			return
 		}
 
 		ownee, ok = tombstone.Obj.(metav1.Object)
 		if !ok {
-			utilruntime.HandleError(fmt.Errorf("Tombstone contained object that is not a metav1 object %#v", obj))
+			utilruntime.HandleError(fmt.Errorf("tombstone contained object that is not a metav1 object %#v", obj))
 			return
 		}
 	}
 
-	if owner := ownerutil.GetOwnerByKind(ownee, v1alpha1.CatalogSourceKind); owner != nil {
-		if err := o.catSrcQueueSet.Requeue(owner.Name, ownee.GetNamespace()); err != nil {
-			o.Log.Warn(err.Error())
+	logger := o.Log.WithFields(logrus.Fields{
+		"sync":      "resourcedeletion",
+		"name":      ownee.GetName(),
+		"namespace": ownee.GetNamespace(),
+	})
+
+	owner := ownerutil.GetOwnerByKind(ownee, v1alpha1.CatalogSourceKind)
+	if owner == nil {
+		logger.Debug("no owner catalogsource found")
+		return
+	}
+
+	logger = logger.WithFields(logrus.Fields{
+		"owner":     owner.Name,
+		"ownerkind": owner.Kind,
+	})
+
+	// Get the owner CatalogSource
+	catsrc, err := o.lister.OperatorsV1alpha1().CatalogSourceLister().CatalogSources(ownee.GetNamespace()).Get(owner.Name)
+	if err != nil {
+		logger.WithError(err).Warn("could not get owner catalogsource from cache")
+		return
+	}
+
+	// Check the registry server
+	checker := o.checker.CheckerForSource(catsrc)
+	healthy, err := checker.CheckRegistryServer(catsrc)
+	if err != nil {
+		logger.WithError(err).Warn("error checking registry health")
+	} else if !healthy {
+		logger.Debug("registry server unhealthy, updating catalog source")
+		catsrc.Status.RegistryServiceStatus = nil
+		_, err = o.client.OperatorsV1alpha1().CatalogSources(catsrc.GetNamespace()).UpdateStatus(catsrc)
```

Comment:
So we just clear out the status, so that the reconciler knows it has work to do?

Reply:
yes